src/index.ts

```diff
@@ -0,0 +1,106 @@
+import net from "net";
+import {BeaconState} from "./types";
+import * as msgs from "./messages";
+import Peer, {PeerOpts} from "./peer";
+
+export interface HobbitsOpts {
+  port?: number,
+  networkId: number,
+  chainId: number,
+  bootnodes?: string[],
+  state: BeaconState
+}
+
+export default class HobbitsP2PNetwork {
+  private port: number;
+  private networkId: number;
+  private chainId: number;
+  private peers: Peer[];
+  private bootnodes: string[];
+  private state: BeaconState;
+  private server: net.Server;
+
+  /**
+   * Constructor
+   * @param {HobbitsOpts} opts
+   */
+  public constructor(opts: HobbitsOpts) {
+    this.port = opts.port || 9000;
+    this.networkId = opts.networkId;
+    this.chainId = opts.chainId;
+    this.state = opts.state;
+    this.peers = [];
+    this.bootnodes = opts.bootnodes || [];
+    this.server = net.createServer();
+  }
+
+  /**
+   * Connects to static peers
+   */
+  private connectStaticPeers = () => {
+    this.bootnodes.map((bootnode: string) => {
+      this.connect(bootnode);
+    });
+    console.log(`Connected to ${this.peers.length} static peers`);
+  };
+
+  /**
+   * Connects to a peer
+   * @param {string} ip
+   */
+  private connect = (ip: string) => {
+    const peer: Peer = new Peer({
+      ip,
+      port: this.port
+    });
+
+    const res: boolean = peer.connect();
+    if (res) {
+      // Add peer to peers list
+      this.peers.push(peer);
+      // Send 0x00 Hello message
+      this.sendHello(peer)
+    }
+  };
+
+  /**
+   * Sends 0x00 message "Hello"
+   * @param peer
+   */
+  private sendHello = (peer) => {
+    const msg: msgs.Hello = {
+      network_id: this.networkId,
+      chain_id: this.chainId,
+      latest_finalized_root: this.state.latestFinalizedRoot,
+      latest_finalized_epoch: this.state.latestFinalizedEpoch,
+      best_root: this.state.latestFinalizedRoot,
+      best_slot: this.state.latestFinalizedSlot
+    };
+    peer.connection.write(msg);
+  };
+
+  /**
+   * Starts the server
+   */
+  public start = () => {
+    this.server.on("connection", (socket) => {
+      console.log(`CONNECTED: ${socket.remoteAddress}:${socket.remotePort}`)
```

Comment:
I think we'll want to create new peers here

Reply:
allow the Peer constructor to pass in a `socket` (or connection, whichever you want to call it)
then here you can do:
```typescript
this.peers.push(new Peer({socket}))
```

If we don't have something like this, we won't be able to connect two `HobbitsP2PNetwork`s with Peers on both sides.